cronjobs.yml

```diff
@@ -9,12 +9,40 @@ cronJobs:
   - name: process-test-mappings
     schedule: "0 1 * * *"
     command: ["/bin/sh", "cronjobs/process_test_mapping_work_items.sh"]
+    resources: # Burstable
+      requests: # guaranteed amount of resources
+        cpu: "10m"
+        memory: "5Mi"
+      limits: # maximum allowed resources
+        cpu: "1.5"
+        memory: "600Mi"
   - name: process-task-mappings
     schedule: "0 2 * * *"
     command: ["/bin/sh", "cronjobs/process_task_mapping_work_items.sh"]
+    resources: # Burstable
+      requests: # guaranteed amount of resources
+        cpu: "10m"
+        memory: "5Mi"
+      limits: # maximum allowed resources
+        cpu: "2.5"
+        memory: "1.5Gi"
   - name: update-test-mappings
     schedule: "0 3 * * *"
     command: ["/bin/sh", "cronjobs/update_test_mappings.sh"]
+    resources: # Burstable
+      requests: # guaranteed amount of resources
+        cpu: "900m"
```

Comment:
Why do the `update` cronjobs need more resources than the `process` cronjobs? Since they only run on 1 day of data, as opposed to 6 months, shouldn't the `updates` use fewer resources?

Reply:
This is explained in the [application resources docs here](https://github.com/10gen/kanopy-docs/blob/f4cbf669a54574420248792ad3c78b9e4fa96f81/docs/application_resources.md).

> Requests are the values that you expect each container to use and is used by the scheduler to place the container (Pod) in the appropiate node given the available resources on it. Limits are a maximum cap of resources that a container can consume. If utilization exceeds the requested values it will cause the container's CPU use to be throttled, or to be evicted with an Out Of Memory (OOM) error. More information about requests and limits can be found in the official docs here.

The update jobs typically use the same amount of resources every time so that is what the requests value is set to ( have a look at [grafana](https://grafana.corp.mongodb.com/d/zenZa6oWk/selected-tests-service?orgId=4&refresh=5s&from=now-7d&to=now)). The limit is set to some theoretical max.

The process jobs typically use very little resources (as they don't do much and are usually so low or so fast that we can't even see some of the resources [grafana](https://grafana.corp.mongodb.com/d/zenZa6oWk/selected-tests-service?orgId=4&refresh=5s&from=now-7d&to=now)) so the requests values are set low. However, when they are doing a full reload they then use a lot of resources so the limit is set quite high.

This seems counter-intuitive to what you would expect, but I think it makes sense.

Reply:
Ah, I see, that makes sense. Thanks for explaining.